table/scanner_test.go

```diff
@@ -458,6 +458,23 @@ func (s *ScannerSuite) TestPartitionedTables() {
 	}
 }
 
+func (s *ScannerSuite) TestNestedColumns() {
+	mem := memory.NewCheckedAllocator(memory.DefaultAllocator)
+	defer mem.AssertSize(s.T(), 0)
+
+	ident := catalog.ToIdentifier("default", "test_all_types")
```

Comment:
is this the nested col?
https://github.com/apache/iceberg-go/blob/b1b432e7164b66a61a00b2a6ed312596661275b1/dev/provision.py#L202

Reply:
that's one of them. There's also the map and array columns which aren't in the struct. All in all there are several nested columns